src/ng/directive/ngModel.js

```diff
@@ -877,11 +880,22 @@ NgModelController.prototype = {
    * See {@link ngModelOptions} for information about what options can be specified
    * and how model option inheritance works.
    *
+   * <div class="alert alert-warning">
+   * **Note:** overriding the model options with this function only affects the
+   * current `ngModelController` and does not propagate the new options to
+   * `ngModelOptions` directives that have initially inherited from it.
```

Comment:
I wonder if the whole notice is irrelevant, as you can only $overrideModelOptions on ngModel, and nested ngModels are definitely "here be dragons" territory. Or should we mention that this doesn't change the options on ngModelOptions that this instance has inherited from?

Reply:
Actually, I think you are right. The inheritance is between `ngModelOptions` (which can be nested). `ngModel` just takes the options of its nearest `ngModuleOptions` and it can overwrite them. Where the corresponding `ngModelOptions` got its values (inherited or not) is not affected (nor is there any reason to assume it would).